File: apps/mobile/native/tauri-plugin-recorder/ios/MysttRecorderPlugin.scaffold.swift
import Foundation

// TODO: Tauri 2 iOS plugin entrypoint.
// This scaffold intentionally mirrors the recorder ledger contract used by the desktop shell.
//
// Required responsibilities:
// - start / pause / resume / stop native recording
// - persist runtime-state.json before background transitions
// - persist sessions.json and recordings/<session-id>/session.json on save
// - keep original audio until upload + hash verification completes
//
// The shared JSON contract lives in:
// /Volumes/mac_dock/github/mystt/packages/audio-core/src/tauri-recorder.ts

final class MysttRecorderPluginScaffold {
  func startRecording() {
    // TODO: wire AVAudioRecorder / AVAudioEngine background recorder.
  }

  func pauseRecording() {
    // TODO
  }

  func resumeRecording() {
    // TODO
  }

  func stopRecording() {
    // TODO
  }
}

File: apps/mobile/native/tauri-plugin-recorder/android/MysttRecorderPlugin.scaffold.kt
package digital.doublejun.mystt.recorder

// TODO: Tauri 2 Android plugin entrypoint.
// This scaffold intentionally mirrors the recorder ledger contract used by the desktop shell.
//
// Required responsibilities:
// - foreground service microphone capture
// - runtime-state.json flush on background/service lifecycle changes
// - sessions.json + recordings/<session-id>/session.json persistence on save
// - original audio retention until upload + hash verification completes
//
// Shared JSON contract:
// packages/audio-core/src/tauri-recorder.ts

class MysttRecorderPluginScaffold {
  fun startRecording() {
    // TODO
  }

  fun pauseRecording() {
    // TODO
  }

  fun resumeRecording() {
    // TODO
  }

  fun stopRecording() {
    // TODO
  }
}
